code/modules/research/experimentor/experimentor.dm

```diff
@@ -0,0 +1,319 @@
+		if(text2num(scantype) == SCANTYPE_DISCOVER)
+			loaded_item = discover(T)
+			ejectItem()
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR assembles the [T.name] into its new, final form!</span>")
+			playsound(src.loc, 'sound/items/rped.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_OBLITERATE)
+			T.reroll()
+			T.adjustflexibility(0-rand(adjusted_flex_cost*2+5, adjusted_flex_cost*3+5))
+			to_chat(usr, "<span class='notice'>The [T.name] is carefully taken apart and reassembled in a brand-new configuration by the tiny manipulators.</span>")
+			playsound(src.loc, 'sound/items/rped.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_RECYCLE)
+			ejectItem(TRUE)
+			playsound(loc, 'sound/effects/splat.ogg', 50, 1)
+			to_chat(usr, "<span class='notice'>The [T.name] is sliced apart by the laser grid and the parts are stored away.</span>")
+			spareparts += 50
+		use_power(750)
+		//If an adjustment takes flexibility to 0, the tech is destroyed.
+		if(istype(loaded_item, /obj/item/unknown_tech))
+			if(T.flexibility == 0)
+				ejectItem(TRUE)
+				to_chat(usr, "<span class='warning'>The [T.name] cannot handle the adjustment and breaks apart!</span>")
+				playsound(loc, 'sound/effects/splat.ogg', 50, 1)
+				criticalfailure()
+		if (spareparts >= 100)
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR beeps and assembles a new prototype from the spare parts reserve!</span>")
+			playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
+			loaded_item = new/obj/item/unknown_tech/proto_tech()
+			spareparts -= 100
+			ejectItem()
+	src.updateUsrDialog()
+	return
+
+/obj/machinery/r_n_d/experimentor/proc/criticalfailure()
+	var/failuretype = rand(1,8)
+	switch(failuretype)
+		if(1)
+			visible_message("<span class='danger'>Nearby people are hit by debris!</span>")
+			for(var/mob/living/m in oview(1, src))
+				m.apply_damage(15,BRUTE,pick("head","chest","groin"))
+		if(2)
+			visible_message("<span class='danger'>The casing cracks, leaking radiation!</span>")
+			for(var/mob/living/m in oview(1, src))
+				m.apply_effect(25,IRRADIATE)
+		if(3)
+			visible_message("<span class='warning'>The destroyed artifact releases toxic waste!</span>")
+			for(var/turf/T in oview(1, src))
+				if(!T.density)
+					var/obj/effect/decal/cleanable/reagentdecal = new/obj/effect/decal/cleanable/greenglow(T)
+					reagentdecal.reagents.add_reagent("radium", 7)
+		if(4)
+			visible_message("<span class='warning'>The air crackles and pulses with energy!</span>")
+			empulse(src.loc, 4, 0)
+		if(5)
+			var/turf/start = get_turf(src)
+			var/mob/M = locate(/mob/living) in view(src, 3)
+			var/turf/MT = get_turf(M)
+			if(MT)
+				visible_message("<span class='danger'>The coolant cells burst violently!</span>")
+				var/obj/item/projectile/magic/fireball/FB = new /obj/item/projectile/magic/fireball(start)
+				FB.original = MT
+				FB.current = start
+				FB.yo = MT.y - start.y
+				FB.xo = MT.x - start.x
+				FB.fire()
+		if(6)
```

Comment:
Not a code comment, but I'm kind of disappointed the experimentor can no longer spawn angry space bears or other mobs like that.

Reply:
You're not wrong, this is a serious issue that requires careful handling.